### PR TITLE
feat: add generated caution text to the messages with corresponding keywords

### PR DESCRIPTION
**Pull request by OEvgeny**
> Fixes #

## Changelog Entry

- Added a content generated badge to AI-generated messages, in PR [#5298](https://github.com/microsoft/BotFramework-WebChat/pull/5298), by [@OEvgeny](https://github.com/OEvgeny)

## Description

This PR adds a content generated badge to messages that are identified as AI-generated content.

## Design

The implementation targets all white-label and both theme variant experiences:
-  In white-label we show a grayed text at the bottom (before references)
- In fluent variant we show a badge before references
- In copilot variant we hide the badge as there is another one on copilot header area

## Specific Changes

- Created `isAIGeneratedActivity.ts` to identify AI-generated content
- Updated `TextContent.tsx`, `MarkdownTextContent.tsx`, and `PlainTextContent.tsx` to include the generated badge
- Modified `ActivityDecorator.module.css` to style the generated badge
- Updated test files to include scenarios with AI-generated content

## -

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] ~I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/component/src/Attachment/Text/TextContent.tsx [hunk 1]
```diff
@@ -1,30 +1,56 @@
...
   activity: WebChatActivity;
   contentType?: string;
   text: string;
 }>;
 
-const TextContent: FC<Props> = memo(({ activity, contentType = 'text/plain', text }: Props) => {
+const generatedBadgeStyle = {
+  '&.webchat__text-content__generated-badge': {
+    fontSize: `var(${CustomPropertyNames.FontSizeSmall})`,
+    color: `var(${CustomPropertyNames.ColorSubtle})`
```
Comment (compulim) [inline packages/component/src/Attachment/Text/TextContent.tsx:24]:
```suggestion
    color: `var(${CustomPropertyNames.ColorSubtle})`,
    fontSize: `var(${CustomPropertyNames.FontSizeSmall})`
```

File: packages/component/src/Attachment/Text/TextContent.tsx [hunk 2]
```diff
@@ -1,30 +1,56 @@
...
 
   return text ? (
     contentType === 'text/markdown' && supportMarkdown ? (
-      <MarkdownTextContent activity={activity} markdown={text} />
+      <MarkdownTextContent activity={activity} markdown={text}>
+        {generatedBadge}
+      </MarkdownTextContent>
     ) : (
-      <PlainTextContent text={text} />
+      <PlainTextContent text={text}>{generatedBadge}</PlainTextContent>
```
Comment (compulim) [inline packages/component/src/Attachment/Text/TextContent.tsx:49]: Should we `<Fragment>` and put the `generatedBadge` below it?

Comment (OEvgeny) [reply packages/component/src/Attachment/Text/TextContent.tsx:49]: We could but we need the same classes, so paddings are preserved. I was thinking it'd be better to reuse classes in the scope of the component.

File: packages/component/src/Attachment/Text/private/isAIGeneratedActivity.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { getOrgSchemaMessage, type WebChatActivity } from 'botframework-webchat-core';
+
+export default function isAIGeneratedActivity(activity: undefined | WebChatActivity) {
+  return !!(activity && getOrgSchemaMessage(activity?.entities || [])?.keywords?.includes('AIGeneratedContent'));
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/isAIGeneratedActivity.ts:4]: In another universe, I am thinking what-if we do these kind of checks like...

```ts
return safeParse(
  array(
    object(),
    includes({
      '@id': '',
      '@type': 'Message',
      keywords: ...
    })
  ),
  activity.entities
).success;
```

Another universe...

**Review by compulim — APPROVED** [review]

**Pull request merged**